### ISSUE TITLE
release 1.14.10

### ISSUE THREAD

**Issue by jangorecki** [Rdatatable/data.table#5709]
We are getting errors on CRAN due to changes in R-devel, therefore we should provide patched version for the moment, rather then trying to push current master. Release of current master can as well follow the patch release, but releasing master should not be blocking patch release. There is still at least 1 pending issue to resolve in master before it will be ready for CRAN.

**Comment by jangorecki** [comment]
https://github.com/Rdatatable/data.table/compare/1.14.8...hotfix1.14.10
this will be useful before release to ensure all items have been covered

**Comment by TysonStanley** [comment]
Thanks, will get to this over the weekend. Appreciate you setting it up. So the changes for is.atomic() and the suppression of warnings is the only changes I'm seeing, which appear to be what is needed based on the CRAN tests. 

A few questions since we are just starting this new phase of having me release to CRAN:

1. Assume we want to do minor updates to `NEWS.md`?
2. Should I make updates to `DESCRIPTION` to add myself as the maintainer to ensure all contact about the submission comes to me?
3. Any additional checks that you recommend (e.g. reverse dependencies) be run in addition to the CI processes already in place? This is such a small release that it may not be necessary, but thought I'd get your thoughts on it.

**Comment by jangorecki** [comment]
There few more things, all related are on the 1.14.9 milestone.

1. yes, mention in news file will be needed
2. yes
3. ideally would be to go through release checklist, even for setting up and understanding the process. It will now be much easier than later on with 1.15.0 because very few things changed so should go very smoothly.

**Comment by TysonStanley** [comment]
Sounds good. I'll do those and review the release checklist in depth to start to get comfortable with it. 

I agree, it's very convenient we get to do a "practice" round with this much smaller release in advance of the larger 1.15 release.

**Comment by tdhock** [comment]
I don't think revdep checks are necessary for patch releases like this.

**Comment by jangorecki** [comment]
I pushed NEWS file. It will still need minor amendment when closing #5695

**Comment by TysonStanley** [comment]
Updated the dsc file in [hotfix1.14.10](https://github.com/Rdatatable/data.table/tree/hotfix1.14.10)

**Comment by TysonStanley** [comment]
@jangorecki looks like we are waiting on a fix for shift before we submit the patch?

**Comment by jangorecki** [comment]
Yes, only shift fix is left and we are good for CRAN submission. At the very last moment it is worth to look at CRAN check results, in case another breaking change in R-devel will suddenly popup, so we can still put the fix at the last moment.

**Comment by TysonStanley** [comment]
I haven't seen any activity on `shift()` yet. I'm still waiting on that before submission.

**Comment by jangorecki** [comment]
Shift was addressed by Michael

**Comment by TysonStanley** [comment]
Thanks, will set up the CRAN submission this weekend.

**Comment by mmaechler** [comment]
> Thanks, will set up the CRAN submission this weekend.

and then it can happen?   
 It does look a bit embarrassing to have **ERROR* on the CRAN check pages for such a long duration..

**Comment by TysonStanley** [comment]
While we are working on the final piece for the patch release, was running all checks (both `R CMD check` and the items in `CRAN_release.cmd`). For `R CMD check`, there are a few notes.

1. The change in maintainer (not a surprise and should be there)
2. Two URLs connected to https://www.seanlahman.com/ are not working
3. HTML checks were not happy. For the HTML checks, here is the output. 

```
* checking HTML version of manual ... NOTE
Found the following HTML validation problems:
IDateTime.html:4:1 (IDateTime.Rd:47): Warning: <link> inserting "type" attribute
IDateTime.html:12:1 (IDateTime.Rd:47): Warning: <script> proprietary attribute "onload"
IDateTime.html:12:1 (IDateTime.Rd:47): Warning: <script> inserting "type" attribute
IDateTime.html:17:1 (IDateTime.Rd:47): Warning: <table> lacks "summary" attribute
IDateTime.html:80:1 (IDateTime.Rd:88): Warning: <table> lacks "summary" attribute
J.html:4:1 (J.Rd:5): Warning: <link> inserting "type" attribute
J.html:12:1 (J.Rd:5): Warning: <script> proprietary attribute "onload"
J.html:12:1 (J.Rd:5): Warning: <script> inserting "type" attribute
J.html:17:1 (J.Rd:5): Warning: <table> lacks "summary" attribute
J.html:41:1 (J.Rd:21): Warning: <table> lacks "summary" attribute
[TRUNCATED]
update_dev_pkg.html:4:1 (update_dev_pkg.Rd:3): Warning: <link> inserting "type" attribute
update_dev_pkg.html:12:1 (update_dev_pkg.Rd:3): Warning: <script> proprietary attribute "onload"
update_dev_pkg.html:12:1 (update_dev_pkg.Rd:3): Warning: <script> inserting "type" attribute
update_dev_pkg.html:17:1 (update_dev_pkg.Rd:3): Warning: <table> lacks "summary" attribute
update_dev_pkg.html:37:1 (update_dev_pkg.Rd:12): Warning: <table> lacks "summary" attribute
```

The other checks produced a few things for us to take note of, but aren't really necessary for the patch release. We can take care of those in the 1.15.0 release.

**Comment by jangorecki** [comment]
If you are detecting new problem on your machine, be sure to confirm them on another one, as it may ended up being local configuration issue. Usually docker is good place to reproduce, other can be VM, rhub builder, or CI machines.

Before going through all checks in release checklist it may be good to resolve last pending items on 1.14.10 milestone as they may affect the outcome of steps in release checklist.

**Comment by TysonStanley** [comment]
I agree. This was mostly to get familiar with the process and wanted to bring up any that may impact that release that didn't look connected to the complex issue. Will try to reproduce and if so, will open an issue.

**Comment by TysonStanley** [comment]
Not reproducing the HTML notes so can ignore for now.

**Comment by jangorecki** [comment]
https://github.com/Rdatatable/data.table/issues/5777 is not being refreshed since 28 Nov so we cannot be sure if it is false positive. We will have to proceed to CRAN submission having this one unresolved for the moment.

**Comment by jangorecki** [comment]
There is still
```
chmatch.c:95:26: warning: argument 1 range [18446744071562067968, 18446744073709551615] exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
chmatch.c:95:26: warning: argument 1 range [18446744071562067968, 18446744073709551615] exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
```
was fixed in master, will cherry pick to hotfix branch in a moment

**Comment by jangorecki** [comment]
Two more rounds of compilation warnings fixes.
We are good to go for CRAN submission.

**Comment by jangorecki** [comment]
Afair to reproduce those locally xml2 pkg is needed. Described in release procedure.

**Comment by TysonStanley** [comment]
Was hoping this was a false positive on Winbuilder bc it was only showing up there but in the other CRAN checks it showed up on Debian too. Any ideas?

```
Flavor: r-devel-linux-x86_64-debian-gcc, r-devel-windows-x86_64
Check: CRAN incoming feasibility, Result: NOTE
  Maintainer: 'Tyson Barrett <t.barrett88@gmail.com>'
  
  New maintainer:
    Tyson Barrett <t.barrett88@gmail.com>
  Old maintainer(s):
    Matt Dowle <mattjdowle@gmail.com>
  
  Size of tarball: 5211907 bytes

Flavor: r-devel-linux-x86_64-debian-gcc, r-devel-windows-x86_64
Check: S3 generic/method consistency, Result: NOTE
  Apparent methods for exported generics not registered:
    as.IDate.IDate
  See section 'Registering S3 methods' in the 'Writing R Extensions'
  manual.

Flavor: r-devel-linux-x86_64-debian-gcc
Check: re-building of vignette outputs, Result: NOTE
  Re-building vignettes had CPU time 3.2 times elapsed time
```

Obviously the first isn't an issue and my guess the vignette one is not a surprise or new. But the second one, any ideas on why that one would only pop up on R-devel for Linux/Windows?

**Comment by TysonStanley** [comment]
Thanks.

* Yes, compressed the .Rraw files but can check that those are what are being built with.
* Can add the method to NAMESPACE (was just surprised it only showed up on R-devel...)
* Do the vignettes not have that already in there for a purpose? (ie should I PR that or just use it for the submission)?

**Comment by ben-schwen** [comment]
> * Can add the method to NAMESPACE (was just surprised it only showed up on R-devel...)

This is note 2 from 1.13.4. Check news but we could try at least

**Comment by jangorecki** [comment]
CRAN checks on that are unreliable. There are reports from users that set limit to 2 but still had those notes (for example, not vignettes). As long as CRAN does not provide reliable way to respect their policies, and as long as these are only vignettes/examples and not users code, then I feel it is better to set it to 1. Just be sure to re-set it at the end. To not change the state of environment that renders vignette.

**Comment by TysonStanley** [comment]
Is the best way to reset it by just running `setDTthreads()`?

**Comment by jangorecki** [comment]
```r
.old.th = setDTthreads(1)
```
...
```r
setDTthreads(.old.th)
```

**Comment by jangorecki** [comment]
Please  include those fixed in hotfix branch here, so branch reflect the version submitted to CRAN. Except for the version bump explained in the procedure.

**Comment by mmaechler** [comment]
> it looks accurate, maybe it's a new check:

Yes, it's very new; I've seen it in a package of mine which has been unchanged for years.

**Comment by eddelbuettel** [comment]
Congrats to all, and especially the new maintainer !!, on getting a new version onto CRAN!  On to the 1.15.* series then?

**Comment by MichaelChirico** [comment]
indeed! trust dirk to be one of the first to spot the release :)

**Comment by jangorecki** [comment]
@TysonStanley still please do tag 1.14.10 and push to git, as explained in release procedure, avoiding even version number in the package code. Only in git tag name we want to have even number.

It used to be happen couple times that after CRAN submission new errors has been detected in CRAN checks, that haven't been detected during submission. Then another patch release is needed (slide 9 from here shows that a bit https://jangorecki.gitlab.io/r-talks/2018-07-03_Wroclaw_What_s-new-in-data.table/What-s-new-in-data.table.pdf).
Then we just fork from 1.14.10 tag, add fixed and push as 1.14.12.

**Comment by TysonStanley** [comment]
Tagged the 1.14.10 release to CRAN with edits that passed CRAN checks (as far as have been run thus far)

https://github.com/Rdatatable/data.table/releases/tag/1.14.10
